### PR TITLE
Address remaining diffs noted #821

### LINKED ISSUE #821: CDash: Failing tests 2020-03-10
Two tests `ablNeutralStat` and `actLineSimple` are failing. These are likely related to PRs #810 and #783 respectively. 

- [First instance](https://my.cdash.org/test/27161977) of `actLineSimple` failure
- [First instance](https://my.cdash.org/test/27271248) of `ablNeutralStat` failure
- PR #810 also introduced [unused variable warnings](https://my.cdash.org/viewBuildError.php?type=1&buildid=1979016)

### PR DESCRIPTION
**Pull request by psakievich**
**Pull-request type:**
- [x] Bug fix 
- [ ] Documentation update
- [ ] Feature enhancement

## Description of the pull-request

Addressing remaining diffs from #810 as noted in #821.  There was a bug in the implementation from changing from a face lag to face elem alg.  Only ordinal 0 was being used in the ipMap for the `slip` branch of the stress model causing diffs at the wall.

It's interesting that this wasn't caught in `ablNeutralEdge` on the CPU runs.

However, we've tested this with `ablNeutralNGPTrilinos` and `ablNeturalStat` and the diffs were resolved for these two cases.  My GPU build of hypre failed so I didn't test the hypre variants but the mechanism of failure should be the same. This should close #821.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [x] Linux
    - [ ] MacOS
  - Compilers 
    - [ ] GCC
    - [ ] LLVM/Clang
    - [ ] Intel compilers
    - [x] NVIDIA CUDA
- [ ] Compiles without warnings
- [ ] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

File: src/LowMachEquationSystem.C [hunk 1]
```diff
@@ -1186,6 +1186,8 @@ MomentumEquationSystem::register_nodal_fields(
       stk::topology::NODE_RANK, "abl_wall_no_slip_wall_func_node_mask");
   double one = 1;
   stk::mesh::put_field_on_mesh(node_mask, *part, 1, &one);
+  node_mask.modify_on_host();
+  node_mask.sync_to_device();
```
Comment (sayerhs) [inline src/LowMachEquationSystem.C:1190]: The bulkdata is not active yet here.

**Comment by sayerhs** [comment]
@psakievich can you provide more details on how you are doing your GPU build of hypre? If spack, what spack commit are you using?

File: src/LowMachEquationSystem.C (hunk 1 above)
Comment (psakievich) [reply src/LowMachEquationSystem.C:1190]: Good point. I'll remove these. I forgot they were there.

**Comment by psakievich** [comment]
@sayerhs yes I was using `spack develop` spack commit `821769c95eea43a202c591058d0966e53c8be4cf`

**Comment by sayerhs** [comment]
OK. That commit should support `hypre@develop+cuda` Can you share what the error you're seeing is?

**Comment by psakievich** [comment]
> OK. That commit should support `hypre@develop+cuda` Can you share what the error you're seeing is?

Yeah I'll ping you on slack after I add hypre back to the environment

**Pull request merged**